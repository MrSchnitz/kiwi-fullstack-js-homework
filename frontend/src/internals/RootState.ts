import {connectRouter} from "connected-react-router";
import {Action, ThunkAction} from "@reduxjs/toolkit";
import {MainApiInterface, MainApiReducer, MainApiSaga,} from "../pages/MainPage/api/MainAPI";
import {all} from "redux-saga/effects"
import {createBrowserHistory} from "history";

export const history = createBrowserHistory();

export interface RootState {
  // API
  mainApiSlice?: MainApiInterface;
  // Containers

  // Router
  router?: any;
}

export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;

export const mainReducers = {
  // API
  mainApiSlice: MainApiReducer,
  // Containers

  // Router
  router: connectRouter(history),
};

/**
 * IMPORT EVERY OTHER NOT SOMEWHERE ELSE INJECTED SAGA
 */
export default function* rootSaga() {
  yield all([MainApiSaga()]);
}
