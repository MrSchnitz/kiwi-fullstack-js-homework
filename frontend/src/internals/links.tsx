import React, { ReactNode } from "react";
import { Airplane, Partners } from "@kiwicom/orbit-components/lib/icons";

export interface LinkInterface {
  title: string;
  url: string;
  icon?: ReactNode;
}

export const Links = {
  mainPage: "/",
  descriptionPage: "/description",
};

export const NavigationLinks: LinkInterface[] = [
  {
    title: "Main",
    url: Links.mainPage,
    icon: <Airplane />,
  },
  {
    title: "Homework Description",
    url: Links.descriptionPage,
    icon: <Partners />,
  },
];
