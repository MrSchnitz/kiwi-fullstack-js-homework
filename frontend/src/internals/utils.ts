export const ENV = process.env.BE || "http://localhost:4000";
