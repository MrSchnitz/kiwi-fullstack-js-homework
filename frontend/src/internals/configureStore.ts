import { routerMiddleware } from "connected-react-router";
import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from "@reduxjs/toolkit";
import rootSaga, { mainReducers } from "./RootState";
import createSagaMiddleware from "redux-saga";
import { history } from "./RootState";

export function configureAppStore() {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with saga middleware and router middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const store = configureStore({
    reducer: combineReducers({ ...mainReducers }),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools: process.env.NODE_ENV !== "production",
  });

  // Add all combined Sagas to middleware
  sagaMiddleware.run(rootSaga);

  return store;
}
