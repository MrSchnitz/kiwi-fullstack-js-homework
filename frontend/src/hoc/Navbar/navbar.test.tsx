import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { withNavigation } from "./navbar.component";
import { configureAppStore } from "../../internals/configureStore";
import { Provider } from "react-redux";

describe("NavBar HOC component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const store = configureAppStore();
    const NavigationComponent = withNavigation(<div />);
    wrapper = shallow(
      <Provider store={store}>
        <NavigationComponent />
      </Provider>
    );
  });

  it("should render NavBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
