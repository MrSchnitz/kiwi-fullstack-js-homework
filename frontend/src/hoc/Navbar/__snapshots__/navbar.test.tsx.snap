// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar HOC component should render NavBar component 1`] = `ShallowWrapper {}`;
